opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1394,6 +1394,11 @@ namespace Opm
         // the result will contain the derivative with resepct to GTotal in segment seg,
         // and the derivatives with respect to WFrac GFrac in segment seg_upwind.
         // the derivative with respect to SPres should be zero.
+        if (seg == 0 && well_type_ == INJECTOR) {
+            const double* distr = well_controls_get_current_distr(well_controls_);
```

Comment:
gas rate will be wrong here.

Reply:
If it is RESV control, then the rate for all the phases will be wrong.

Reply:
I am working on fixing this. I will give you a hint when it is ready. No need for testing from your side before this is fixed. I need to assure that everything is scaled with the same factor. Do you have a small test case for MSW injectors that you could send me?

Reply:
> Do you have a small test case for MSW injectors that you could send me?

No. On the other hand, smaller test for MSW injectors might not need this PR to fix, since it is difficult to produce the complicated situation need to do the upwinding.  For the producers, we saw one single case gives strong sign that upwinding is required for that report step.

Reply:
Fixed.